### ISSUE TITLE
Token Not Being Passed Into Contained Action Setup-Python

### ISSUE THREAD

**Issue by HaleenUptain** [advanced-security/policy-as-code#11]
**Describe the bug**
We are testing the new policy-as-code action on GHES v3.7.1 and running into a API rate limit exceeded issue due to a change in the way python is installed.  The python install functionality in the previous action (https://github.com/GeekMasher/advanced-security-compliance) worked and is different from this new action's python install implementation (https://github.com/advanced-security/policy-as-code) which is not working.  We also notice the token we pass in is not being passed down into the setup-python action.  Shouldn't it be?  How do we address this API rate limit issue?

**To Reproduce**
We have a simple workflow setup to perform a java maven build for our hello world application and then it runs the compliance job.  Here is a sanitized version of our workflow file.  For some reason the file would not upload as zip, txt, or yml so pasting in verbatim here:
name: "CodeQL"

on:
  push:
    branches: one_error
  workflow_dispatch:

jobs:
  analyze:
    name: Analyze
    runs-on: [ ***sanitized data*** ]
    container:
      image: ***sanitized data***  
    env:
      MAVEN_CLI_OPTS: "--batch-mode --errors --fail-at-end --show-version -DinstallAtEnd=true -DdeployAtEnd=true -DskipTests -DskipFormat"
    permissions:
      actions: read
      contents: read
      security-events: write

    strategy:
      fail-fast: false
      matrix:
        language: [ 'java' ]

    steps:
    - name: Checkout repository
      uses: actions/checkout@v3

    - name: Initialize CodeQL
      uses: github/codeql-action/init@v2
      with:
        languages: ${{ matrix.language }}
        queries: security-extended,security-and-quality
        
    - name: Build with Maven
      run: |
          mkdir -p .cibuild
          mvn -X -s settings.xml help:evaluate -Dexpression=project.version | grep -v '\[' > .cibuild/version
          mvn -e -X -s settings.xml $MAVEN_CLI_OPTS -D***sanitized data*** -D*** sanitized data*** clean package
           
    - name: Perform CodeQL Analysis
      uses: github/codeql-action/analyze@v2

  compliance:
    runs-on: [ ***sanitized data*** ]
    
    needs: [ analyze ]
    
    steps:
    - name: Advance Security Compliance Action
      uses: advanced-security/policy-as-code@v2.1
      with:
        # The owner/repo of where the policy is stored
        policy: ***sanitized data***
        policy-path: policy/security-policy.yml
        policy-branch: main
        
        token: ${{ secrets.ACCESS_TOKEN }}

        action: break

        argvs: '--disable-dependabot --disable-secret-scanning --disable-dependency-licensing --disable-dependencies --disable-caching'

**Expected behavior**
The compliance action runs successfully without encountering a Rate Limit error.

**Job Log**
Here is a snip from the job log that points to what is happening at the time the failure is encountered.  It is at the stage where it is trying to install python.

2023-02-08T14:53:32.2732411Z ##[group]Run actions/setup-python@v2
2023-02-08T14:53:32.2732690Z with:
2023-02-08T14:53:32.2732948Z   python-version: 3.10
2023-02-08T14:53:32.2733677Z   token: ***
2023-02-08T14:53:32.2733905Z ##[endgroup]
2023-02-08T14:53:32.4246659Z Version 3.10 was not found in the local cache
2023-02-08T14:53:32.5690088Z ##[error]API rate limit exceeded for 170.40.176.1. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)

**Comment by GeekMasher** [comment]
Thanks for sharing @HaleenUptain, it looks like to me this might be a rate limiting issue 🤔 I have added it the `develop` branch a quick fix to help with rate limiting.

```yaml
- name: Advance Security Policy as Code
  uses: advanced-security/policy-as-code@develop
```

Over 5k requests in under an hours sounds like a lot, is this token set at the organisation level and used across multiple repos or is it the Actions token?

I would recommend if you are hitting the 5k limit with that token and if its a token at an organization level to use a GitHub App for this as it has x3 the limit.

**Comment by HaleenUptain** [comment]
@GeekMasher, we switched our test to use develop branch and proved out your changed resolved the API rate limit issue, we do not see that issue now.
  
Regarding the token, we are using a Personal API token and have that set up as a action secret. 

Will you be releasing a new version of the action that has this fix?  Assumed the develop branch was just temporary and test for proving out your change worked.

**Comment by GeekMasher** [comment]
@HaleenUptain Sorry for the delay but the issue with rate limiting should now be fixed in v2.2 with #15. I'll leave this open until you have confirmed the issue is resolved

**Comment by GeekMasher** [comment]
@HaleenUptain can you please let me know if the version in the previous comment fixed this issue for you. Otherwise I will auto-close the issue next week
